File: step_definitions/Verify ability to enter new shipping and billing addresses.java
package com.cucumber.steps;

import com.page_objects.CheckoutPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class AddressSetupSteps extends CheckoutPage {

    @Given("a user account is logged in")
    public void userAccountIsLoggedIn() {
        loginUser();
    }

    @When("the user navigates to the checkout page")
    public void userNavigatesToCheckoutPage() {
        navigateToCheckoutPage();
    }

    @Then("the checkout page is displayed with options to enter shipping and billing addresses")
    public void checkoutPageDisplayedWithAddressOptions() {
        verifyCheckoutPageDisplayed();
    }

    @Given("the user is on the checkout page")
    public void userOnCheckoutPage() {
        verifyCheckoutPageDisplayed();
    }

    @When("the user clicks on 'Add New Address' for shipping")
    public void userClicksAddNewAddressForShipping() {
        clickAddNewShippingAddress();
    }

    @Then("a form to enter new shipping address is displayed")
    public void formToEnterNewShippingAddressDisplayed() {
        verifyShippingAddressFormDisplayed();
    }

    @Given("the shipping address form is displayed")
    public void shippingAddressFormDisplayed() {
        verifyShippingAddressFormDisplayed();
    }

    @When("the user enters shipping address details {string}")
    public void userEntersShippingAddressDetails(String addressDetails) {
        enterShippingAddressDetails(addressDetails);
    }

    @When("enters phone number {string}")
    public void userEntersPhoneNumber(String phoneNumber) {
        enterPhoneNumber(phoneNumber);
    }

    @Then("the shipping address details are entered correctly")
    public void shippingAddressDetailsEnteredCorrectly() {
        verifyShippingAddressDetailsEntered();
    }

    @Given("the user has entered shipping address details")
    public void userHasEnteredShippingAddressDetails() {
        verifyShippingAddressDetailsEntered();
    }

    @When("the user clicks on 'Add New Address' for billing")
    public void userClicksAddNewAddressForBilling() {
        clickAddNewBillingAddress();
    }

    @Then("a form to enter new billing address is displayed")
    public void formToEnterNewBillingAddressDisplayed() {
        verifyBillingAddressFormDisplayed();
    }

    @Given("the billing address form is displayed")
    public void billingAddressFormDisplayed() {
        verifyBillingAddressFormDisplayed();
    }

    @When("the user enters billing address details {string}")
    public void userEntersBillingAddressDetails(String billingDetails) {
        enterBillingAddressDetails(billingDetails);
    }

    @When("enters email {string}")
    public void userEntersEmail(String email) {
        enterEmail(email);
    }

    @Then("billing address details are entered correctly")
    public void billingAddressDetailsEnteredCorrectly() {
        verifyBillingAddressDetailsEntered();
    }

    @Given("the user has entered shipping and billing address details")
    public void userHasEnteredShippingAndBillingAddressDetails() {
        verifyShippingAndBillingDetailsEntered();
    }

    @When("the user clicks 'Save' to store the new addresses")
    public void userClicksSaveToStoreNewAddresses() {
        clickSaveAddresses();
    }

    @Then("addresses are saved successfully without errors")
    public void addressesSavedSuccessfullyWithoutErrors() {
        verifyAddressesSavedSuccessfully();
    }

    @Given("the addresses are saved successfully")
    public void addressesSavedSuccessfully() {
        verifyAddressesSavedSuccessfully();
    }

    @When("the user views the list of saved addresses")
    public void userViewsListOfSavedAddresses() {
        viewSavedAddresses();
    }

    @Then("newly entered addresses appear in the list of saved addresses")
    public void newlyEnteredAddressesAppearInList() {
        verifyNewAddressesInList();
    }

    @Given("the newly entered addresses are saved")
    public void newlyEnteredAddressesSaved() {
        verifyAddressesSavedSuccessfully();
    }

    @When("the user proceeds to checkout using the new addresses")
    public void userProceedsToCheckoutUsingNewAddresses() {
        proceedToCheckoutWithNewAddresses();
    }

    @Then("checkout process continues using the newly entered addresses")
    public void checkoutProcessContinuesUsingNewAddresses() {
        verifyCheckoutProcessWithNewAddresses();
    }

    @Given("the checkout process continues using the new addresses")
    public void checkoutProcessContinuesUsingNewAddresses() {
        verifyCheckoutProcessWithNewAddresses();
    }

    @When("the checkout is confirmed")
    public void checkoutIsConfirmed() {
        confirmCheckout();
    }

    @Then("a confirmation message is displayed indicating the use of new addresses")
    public void confirmationMessageDisplayedForNewAddresses() {
        verifyConfirmationMessageForNewAddresses();
    }

    @Given("the user is entering addresses")
    public void userIsEnteringAddresses() {
        startEnteringAddresses();
    }

    @When("the addresses are entered")
    public void addressesAreEntered() {
        enterAddresses();
    }

    @Then("no error messages are displayed; addresses are accepted")
    public void noErrorMessagesDisplayedAddressesAccepted() {
        verifyNoErrorMessages();
    }

    @Then("address format is validated and correct")
    public void addressFormatValidatedAndCorrect() {
        verifyAddressFormat();
    }

    @Then("all required fields are populated correctly")
    public void allRequiredFieldsPopulatedCorrectly() {
        verifyRequiredFieldsPopulated();
    }

    @When("addresses in different valid formats are entered")
    public void addressesInDifferentValidFormatsEntered() {
        enterAddressesInDifferentFormats();
    }

    @Then("addresses in different valid formats are accepted")
    public void addressesInDifferentFormatsAccepted() {
        verifyAddressesInDifferentFormatsAccepted();
    }

    @When("international addresses are entered")
    public void internationalAddressesEntered() {
        enterInternationalAddresses();
    }

    @Then("international addresses are entered and accepted correctly")
    public void internationalAddressesAcceptedCorrectly() {
        verifyInternationalAddressesAccepted();
    }
}

File: step_definitions/Verify ability to enter new shipping and billing addresses.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import CheckoutPage from "../pages/checkoutPage";

Given("a user account is logged in", () => {
 cy.login();
});

When("the user navigates to the checkout page", () => {
 CheckoutPage.visit();
});

Then("the checkout page is displayed with options to enter shipping and billing addresses", () => {
 CheckoutPage.verifyPageDisplayed();
});

Given("the user is on the checkout page", () => {
 CheckoutPage.verifyPageDisplayed();
});

When("the user clicks on 'Add New Address' for shipping", () => {
 CheckoutPage.clickAddNewShippingAddress();
});

Then("a form to enter new shipping address is displayed", () => {
 CheckoutPage.verifyShippingFormDisplayed();
});

Given("the shipping address form is displayed", () => {
 CheckoutPage.verifyShippingFormDisplayed();
});

When("the user enters shipping address details {string}", (address) => {
 CheckoutPage.enterShippingAddress(address);
});

And("enters phone number {string}", (phone) => {
 CheckoutPage.enterShippingPhoneNumber(phone);
});

Then("the shipping address details are entered correctly", () => {
 CheckoutPage.verifyShippingAddressEntered();
});

Given("the user has entered shipping address details", () => {
 CheckoutPage.verifyShippingAddressEntered();
});

When("the user clicks on 'Add New Address' for billing", () => {
 CheckoutPage.clickAddNewBillingAddress();
});

Then("a form to enter new billing address is displayed", () => {
 CheckoutPage.verifyBillingFormDisplayed();
});

Given("the billing address form is displayed", () => {
 CheckoutPage.verifyBillingFormDisplayed();
});

When("the user enters billing address details {string}", (address) => {
 CheckoutPage.enterBillingAddress(address);
});

And("enters email {string}", (email) => {
 CheckoutPage.enterBillingEmail(email);
});

Then("billing address details are entered correctly", () => {
 CheckoutPage.verifyBillingAddressEntered();
});

Given("the user has entered shipping and billing address details", () => {
 CheckoutPage.verifyShippingAddressEntered();
 CheckoutPage.verifyBillingAddressEntered();
});

When("the user clicks 'Save' to store the new addresses", () => {
 CheckoutPage.clickSaveAddresses();
});

Then("addresses are saved successfully without errors", () => {
 CheckoutPage.verifyAddressesSavedSuccessfully();
});

Given("the addresses are saved successfully", () => {
 CheckoutPage.verifyAddressesSavedSuccessfully();
});

When("the user views the list of saved addresses", () => {
 CheckoutPage.viewSavedAddresses();
});

Then("newly entered addresses appear in the list of saved addresses", () => {
 CheckoutPage.verifyNewAddressesInList();
});

Given("the newly entered addresses are saved", () => {
 CheckoutPage.verifyAddressesSavedSuccessfully();
});

When("the user proceeds to checkout using the new addresses", () => {
 CheckoutPage.proceedToCheckoutWithNewAddresses();
});

Then("checkout process continues using the newly entered addresses", () => {
 CheckoutPage.verifyCheckoutProcessContinues();
});

Given("the checkout process continues using the new addresses", () => {
 CheckoutPage.verifyCheckoutProcessContinues();
});

When("the checkout is confirmed", () => {
 CheckoutPage.confirmCheckout();
});

Then("a confirmation message is displayed indicating the use of new addresses", () => {
 CheckoutPage.verifyConfirmationMessageDisplayed();
});

Given("the user is entering addresses", () => {
 CheckoutPage.startEnteringAddresses();
});

When("the addresses are entered", () => {
 CheckoutPage.enterAddresses();
});

Then("no error messages are displayed; addresses are accepted", () => {
 CheckoutPage.verifyNoErrorMessages();
});

Given("the user is entering addresses", () => {
 CheckoutPage.startEnteringAddresses();
});

When("the addresses are entered", () => {
 CheckoutPage.enterAddresses();
});

Then("address format is validated and correct", () => {
 CheckoutPage.verifyAddressFormatValidated();
});

Given("the user is entering addresses", () => {
 CheckoutPage.startEnteringAddresses();
});

When("the addresses are entered", () => {
 CheckoutPage.enterAddresses();
});

Then("all required fields are populated correctly", () => {
 CheckoutPage.verifyRequiredFieldsPopulated();
});

Given("the user is entering addresses", () => {
 CheckoutPage.startEnteringAddresses();
});

When("addresses in different valid formats are entered", () => {
 CheckoutPage.enterDifferentValidFormats();
});

Then("addresses in different valid formats are accepted", () => {
 CheckoutPage.verifyDifferentFormatsAccepted();
});

Given("the user is entering addresses", () => {
 CheckoutPage.startEnteringAddresses();
});

When("international addresses are entered", () => {
 CheckoutPage.enterInternationalAddresses();
});

Then("international addresses are entered and accepted correctly", () => {
 CheckoutPage.verifyInternationalAddressesAccepted();
});